lib/spack/spack/test/database.py

```diff
@@ -285,140 +135,286 @@ def _check_remove_and_add_package(install_db, spec):
     install_db._check_ref_counts()
 
 
-def test_060_remove_and_add_root_package(database):
-    install_db = database.mock.db
-    _check_remove_and_add_package(install_db, 'mpileaks ^mpich')
-
-
-def test_070_remove_and_add_dependency_package(database):
-    install_db = database.mock.db
-    _check_remove_and_add_package(install_db, 'dyninst')
-
-
-def test_080_root_ref_counts(database):
-    install_db = database.mock.db
-    rec = install_db.get_record('mpileaks ^mpich')
-
-    # Remove a top-level spec from the DB
-    install_db.remove('mpileaks ^mpich')
-
-    # record no longer in DB
-    assert install_db.query('mpileaks ^mpich', installed=any) == []
-
-    # record's deps have updated ref_counts
-    assert install_db.get_record('callpath ^mpich').ref_count == 0
-    assert install_db.get_record('mpich').ref_count == 1
-
-    # Put the spec back
-    install_db.add(rec.spec, spack.store.layout)
-
-    # record is present again
-    assert len(install_db.query('mpileaks ^mpich', installed=any)) == 1
-
-    # dependencies have ref counts updated
-    assert install_db.get_record('callpath ^mpich').ref_count == 1
-    assert install_db.get_record('mpich').ref_count == 2
-
+def _mock_install(spec):
+    s = spack.spec.Spec(spec)
+    s.concretize()
+    pkg = spack.repo.get(s)
+    pkg.do_install(fake=True)
 
-def test_090_non_root_ref_counts(database):
-    install_db = database.mock.db
 
-    install_db.get_record('mpileaks ^mpich')
-    install_db.get_record('callpath ^mpich')
+def _mock_remove(spec):
+    specs = spack.store.db.query(spec)
+    assert len(specs) == 1
+    spec = specs[0]
+    spec.package.do_uninstall(spec)
 
-    # "force remove" a non-root spec from the DB
-    install_db.remove('callpath ^mpich')
 
-    # record still in DB but marked uninstalled
-    assert install_db.query('callpath ^mpich', installed=True) == []
-    assert len(install_db.query('callpath ^mpich', installed=any)) == 1
+@pytest.mark.db
```

Comment:
Why move everything into a class?  Is this necessary?  I generally prefer for the `pytest` tests that they *not* be in `unittest` style classes... it's generally cleaner and I thought we were moving this way.

For module-level marks, you can do this:

```python
pytestmark = pytest.mark.db
```

Reply:
> Why move everything into a class? Is this necessary? I generally prefer for the pytest tests that they not be in unittest style classes...

They're not really unittest like classes (the lifetime of fixtures can still be selected). I was doing this to move common markers at the class level, but I see your point and will use `pytestmark` here.